> Does this PR need to also handle
> 
> > error MSB4018: (NETCORE_ENGINEERING_TELEMETRY=Build) The "RepoTasks.RemoveSharedFrameworkDependencies" task failed unexpectedly.
> > System.MissingMethodException: Method not found: 'Boolean System.Collections.Generic.ISet`1.Contains(!0)'.    at RepoTasks.RemoveSharedFrameworkDependencies.FilterDependencies(String targetPath, ISet`1 dependencyToRemove)
> > at RepoTasks.RemoveSharedFrameworkDependencies.Execute() in /_/src/aspnetcore/eng/tools/RepoTasks/RemoveSharedFrameworkDependencies.cs:line 38
> > at Microsoft.Build.BackEnd.TaskExecutionHost.Microsoft.Build.BackEnd.ITaskExecutionHost.Execute()
> > at Microsoft.Build.BackEnd.TaskBuilder.ExecuteInstantiatedTask(ITaskExecutionHost taskExecutionHost, TaskLoggingContext taskLoggingContext, TaskHost taskHost, ItemBucket bucket, TaskExecutionMode howToExecuteTask)
> 
> Which is also occurring in the blocked installer build: https://github.com/dotnet/installer/pull/19577/checks?check_run_id=24250325560
> 
> From the other PR, it sounds like this is due to project not use the shared framework workaround and so its compiling against an out of date BCL?

I think you are right, we also need a change with the build infrastructure change. I'm not familiar with that area so I think we'll need @wtgodbe for that. I'm not sure if we can separate that change into its own change that runs with the current SDK so that we can merge it and make things flow, get a new installer and then update other PRs with it.

--- a/eng/tools/RepoTasks/RemoveSharedFrameworkDependencies.cs
+++ b/eng/tools/RepoTasks/RemoveSharedFrameworkDependencies.cs
@@ -64,11 +64,16 @@ public class RemoveSharedFrameworkDependencies : Microsoft.Build.Utilities.Task
                     var updatedGroup = new PackageDependencyGroup(group.TargetFramework, packages);
                     foreach (var dependency in group.Packages)
                     {
-                        if (dependencyToRemove.Contains(dependency.Id))
+                        // Hack to temporarily avoid using ISet.Contains
+                        if (!dependencyToRemove.Add(dependency.Id))
                         {
                             referencesFrameworkOnlyAssembly = true;
                             Log.LogMessage($"  Remove dependency on '{dependency.Id}'");
                             continue;
+                        }
+                        else
+                        {
+                            dependencyToRemove.Remove(dependency.Id);
                         }
 
                         packages.Add(dependency);

--- a/src/Identity/Specification.Tests/src/UserManagerSpecificationTests.cs
+++ b/src/Identity/Specification.Tests/src/UserManagerSpecificationTests.cs
@@ -644,9 +644,9 @@ public abstract class UserManagerSpecificationTestBase<TUser, TKey>
         var logins = await manager.GetLoginsAsync(user);
         Assert.NotNull(logins);
         Assert.Single(logins);
-        Assert.Equal(login.LoginProvider, logins[logins.Count - 1].LoginProvider);
-        Assert.Equal(login.ProviderKey, logins[logins.Count - 1].ProviderKey);
-        Assert.Equal(login.ProviderDisplayName, logins[logins.Count - 1].ProviderDisplayName);
+        Assert.Equal(login.LoginProvider, logins[^1].LoginProvider);
+        Assert.Equal(login.ProviderKey, logins[^1].ProviderKey);
+        Assert.Equal(login.ProviderDisplayName, logins[^1].ProviderDisplayName);
         var stamp = await manager.GetSecurityStampAsync(user);
         IdentityResultAssert.IsSuccess(await manager.RemoveLoginAsync(user, login.LoginProvider, login.ProviderKey));
         Assert.Null(await manager.FindByLoginAsync(login.LoginProvider, login.ProviderKey));

